src/common/components/clusterDetail/ProgressBarTexts.tsx

```diff
@@ -12,45 +12,39 @@ type HostProgressProps = {
 };
 
 export const ProgressBarTexts = ({ hosts, hostRole }: HostProgressProps) => {
-  const filteredHosts = hosts.filter((host) => host.role && hostRole === host.role);
-  const failedHostsCount = filteredHosts.filter((host) => host.status === 'error').length;
-  const hostCountText = (hostRole: HostRole) =>
+  const hostRoleText = pluralize(hosts.length, hostRole === 'master' ? 'Control Plane' : 'Worker');
+  const failedHostsCount = hosts.filter((host) => host.status === 'error').length;
+
+  const hostCountText =
```

Comment:
I can only suggest a small change, to remove some duplicate code.

```
  const hostCountText = `${hosts.length} ${pluralize(
    hosts.length,
    hostRole === 'master' ? 'control plane node' : 'worker',
  )}`
```

then 
```
  if (hosts.some((host) => ['cancelled', 'error'].includes(host.status))) {
    const failedHostsCount = hosts.filter((host) => host.status === 'error').length;
    return (
      <ClusterProgressItem icon={<ExclamationCircleIcon color={dangerColor.value} />}>
        <>
          {hostRoleText}
          <br />
          <small>{failedHostsCount}/{hostCountText} failed</small>
        </>
      </ClusterProgressItem>
    );
  }
```

Reply:
Changed, thanks for the suggestion!